### ISSUE TITLE
Update memdb examples

### ISSUE THREAD

**Issue by baal-lgln** [kappa-db/kappa-core#26]
The memdb project repo states that it is superseded by [level-mem](https://github.com/Level/mem) which itself will soon be deprecated because it is superseded by [memory-level](https://github.com/Level/memory-level).

Is there any way to update kappa-core (and the examples) to use memory-level?
Or is this project dead and can anybody lead me to a good alternative?

**Comment by hackergrrl** [comment]
Hi @baal-lgln. Do you have some reason want to move away from `memdb`?

`memdb` continues to work fine, and there aren't any open security vulnerabilities. I don't see a reason to use a new module just for the sake of "it is newer".

**Comment by baal-lgln** [comment]
I can't confirm that there are no open security vulnerabilites. npm audit shows this for me:
```
semver  <5.7.2
Severity: high
semver vulnerable to Regular Expression Denial of Service - https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
No fix available
node_modules/levelup/node_modules/semver
node_modules/semver
  levelup  0.9.0 - 1.3.9
  Depends on vulnerable versions of semver
  node_modules/levelup
    memdb  *
    Depends on vulnerable versions of levelup
    node_modules/memdb
```

memdb itself depends on levelup which itself has also been deprecated because it is superseded by [abstract-level](https://github.com/Level/abstract-level) and has therefore not seen an update in three years.
We can't use dependencies that are failing our security pipelines and packages that depend on deprecated software (which depend on even more deprecated packages) will sooner or later become a security risk.

**Comment by hackergrrl** [comment]
> Versions of the package semver before 7.5.2 on the 7.x branch, before 6.3.1 on the 6.x branch, and all other versions before 5.7.2 are vulnerable to Regular Expression Denial of Service (ReDoS) via the function new Range, **when untrusted user data is provided as a range**.

This doesn't seem to be case with [levelup's use of semver](https://github.com/Level/levelup/blob/1.x/lib/leveldown.js#L17). It's using values from two local `package.json` files.

I'm also less concerned because memdb is being used for examples that folx run on their local machine; I don't expect people to DoS themselves.

I do care about security vulnerabilities, but some of these advisories just seem... kind of silly, given the ways that kappa-core is used. This issue you linked to seems like something that a big corporation running many many servers, with many enemies, might be concerned with.

That said, I'll accept any patches to upgrade to newer versions without security vulnerabilities, if they run ok, but in general I'm skeptical of the endless upgrade hamster wheel.

**Comment by hackergrrl** [comment]
> Is there any way to update kappa-core (and the examples) to use memory-level?

Yes, I'm sure it's possible, but I haven't looked into it specifically.

>  Or is this project dead and can anybody lead me to a good alternative?

I can't speak to "dead" (since I'm here responding!) but I would not say that this stack is being actively maintained.

**Comment by telamon** [comment]
@baal-lgln I'd probably dismiss the output of npm-audit, it's mostly designed to keep servers alive.
I only hacked briefly with kappa, but abstract-level has a quite a small/simple API, so i don't think it would take much to edit the calls. 
Can you see which module depends on mem-db?
